### PR TITLE
Flyte Add Enable/Disable Options for Agent Services

### PR DESCRIPTION
**Pull request by Future-Outlier**
## TL;DR
- Added options to enable (default) or disable the Flyte agent.

## Describe your changes
@pingsutw and I have introduced two options for users to configure the Flyte agent - it can now be either enabled or disabled.

We believe this enhancement will offer more flexibility and cater to diverse user needs and deployment scenarios.

After a thorough consideration, we have decided to set the "enable agent" option as the default setting.

### Enable agent service (Default)
Enable the agent deployment by updating helm and kustomization.
You can use my image on the docker hub to test.
```
cd flyte/docker/sandbox-bundled/
make build
docker tag flyte-sandbox:latest futureoutlier/flyte-sandbox:flyteagent-v2
docker push futureoutlier/flyte-sandbox:flyteagent-v2
```

```
./bin/flytectl demo start --image futureoutlier/flyte-sandbox:flyteagent-v2
```
### Disable agent service (Use flag)
```
./bin/flytectl demo start --image futureoutlier/flyte-sandbox:flyteagent-v2 --disable-agent
```

## Tracking issue
https://github.com/flyteorg/flyte/issues/3936
https://github.com/flyteorg/flytectl/pull/420

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Screen Shot
- **New Pod Addition**: The newly introduced pod is named "flyteagent". You can view the snapshot of its presence here.
image

- **Config Map Update**: The "flyte-sandbox-config" config map situated in the "flyte" namespace has been updated.
Its current snapshot can be viewed here.
image

Notably, there's an addition of the `enabled-plugins` key, which can be viewed here.
image

### CODE REVIEW

File: docker/sandbox-bundled/kustomize/complete-agent/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+helmGlobals:
+  chartHome: ../../../../charts
+helmCharts:
+- name: flyte-sandbo  x
```
Comment (pingsutw) [inline docker/sandbox-bundled/kustomize/complete-agent/kustomization.yaml:4]:
```suggestion
- name: flyte-sandbox
```

File: docker/sandbox-bundled/manifests/complete-agent.yaml [hunk 2]
```diff
@@ -0,0 +1,1902 @@
...
+        disabled: false
+  001-plugins.yaml: |
+    tasks:
+      task-plugins:
+        default-for-task-types:
+          container: container
+          container_array: k8s-array
+          sidecar: sidecar
+        enabled-plugins:
+        - container
```
Comment (pingsutw) [inline docker/sandbox-bundled/manifests/complete-agent.yaml:459]: Need to add agent-service here

Comment (Future-Outlier) [reply docker/sandbox-bundled/manifests/complete-agent.yaml:459]: no problem, thanks a lot!

File: docker/sandbox-bundled/manifests/complete.yaml [hunk 3]
```diff
@@ -448,6 +448,7 @@ data:
         - container
         - sidecar
         - k8s-array
+        - agent-service
```
Comment (pingsutw) [inline docker/sandbox-bundled/manifests/complete.yaml:451]: we need to update the agent service config as well. 
```  agent-service:
    defaultAgent:
      endpoint: "dns:///flyteagent.flyte.svc.cluster.local:8000"
      insecure: true

Comment (Future-Outlier) [reply docker/sandbox-bundled/manifests/complete.yaml:451]: No problem!
Thanks a lot for the review!

**Comment by Future-Outlier** [comment]
Hi reviewers, I think @pingsutw's suggestion is really helpful!
However, as a entry-level user, I think if we make the **key agent-service** from 
```
agent-service:
  defaultAgent:
    endpoint: "dns:///flyteagent.flyte.svc.cluster.local:8000"
      insecure: true
```

to 
```
agent-service:
    defaultAgent:
      endpoint: "dns:///flyteagent.flyte.svc.cluster.local:8000"
      insecure: true
      timeouts:
        GetTask: 100s
      defaultTimeout: 100s
    agents:
      custom_agent:
        endpoint: "dns:///flyteagent.flyte.svc.cluster.local:8001"
        insecure: false
        defaultServiceConfig: '{"loadBalancingConfig": [{"round_robin":{}}]}'
        timeouts:
          GetTask: 100s
        defaultTimeout: 100s
    agentForTaskTypes:
      # It will override the default agent for custom_task, which means propeller will send the request to this agent.
      - custom_task: custom_agent
     ```
     
     will be more user-friendly!
    Please tell me your opinion, thanks a lo!

**Comment by Future-Outlier** [comment]
After a discussion with pingsu, we decide to left
```
agent-service:
  defaultAgent:
    endpoint: "dns:///flyteagent.flyte.svc.cluster.local:8000"
    insecure: true
    timeouts:
      GetTask: 10s
    defaultTimeout: 10s
```
like this!
And it will be far more user friendly!

**Review by pingsutw — APPROVED** [review]

**Comment by Future-Outlier** [comment]
@pingsutw  @jeevb Please take a look, and approve the PR here: https://github.com/flyteorg/flytectl/pull/420
Thanks a lot!

**Review by pingsutw — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
